lucene/classification/src/java/org/apache/lucene/classification/utils/NearestFuzzyQuery.java

```diff
@@ -327,4 +329,9 @@ private boolean equalsTo(NearestFuzzyQuery other) {
         Objects.equals(fieldVals, other.fieldVals);
   }
 
+  @Override
+  public void visit(QueryVisitor visitor, Predicate<String> fieldSelector) {
```

Comment:
what's the advantage over asking the visitor if the field is accepted? I also wonder if it can't just be an implementation detail of a specific visitor if a field is of interest or not? It can be a simple delegate?

Reply:
As in, add an `acceptsField` method to QueryVisitor?  What I'm trying to do is to avoid heavy computations in queries like TermInSetQuery when the visitor isn't actually interested in the terms from that query's field.

Reply:
> As in, add an acceptsField method to QueryVisitor?

yes that's what I mean. I am just trying to think what we would do if we'd have another predicate in the future?

Reply:
++, will change